integration/client/container_test.go

```diff
@@ -2352,3 +2353,121 @@ func TestTaskSpec(t *testing.T) {
 	}
 	<-statusC
 }
+
+func TestContainerAccessMount(t *testing.T) {
+	// The Linux busybox image has a www-data user.
+	username := "www-data"
+	isWindows := runtime.GOOS == "windows"
+	if isWindows {
+		username = "ContainerUser"
+	}
+
+	checkContainerAccessMount(t, username, false)
+}
+
+func TestContainerAccessMountRoot(t *testing.T) {
+	username := "root"
+	if runtime.GOOS == "windows" {
+		username = "ContainerAdministrator"
+	}
+	checkContainerAccessMount(t, username, true)
+}
+
+func checkContainerAccessMount(t *testing.T, username string, runsAsAdmin bool) {
+	t.Parallel()
+
+	client, err := newClient(t, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	var (
+		image       Image
+		ctx, cancel = testContext(t)
+		id          = t.Name()
+	)
+	defer cancel()
+
+	tempDir, err := os.MkdirTemp("", id)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !runsAsAdmin {
+		// If the container we're creating does not run as root/ContainerAdministrator
+		// then allow everyone to read the contents of the folder we'll be mounting as
+		// a volume. On linux, the lack of +x on the source folder, for "others" will mean
+		// we will not be able to enter the mount. On Windows, if we don't grant
+		// GENERIC_READ to "Everyone" an unprivileged user will not be able to read
+		// the contents of the volume mount
+		if err := grantReadToEveryone(tempDir); err != nil {
+			t.Fatal(err)
+		}
+	}
+	defer func() {
+		os.RemoveAll(tempDir)
+	}()
```

Comment:
nit: can't this be simplified to a one line `defer` like many others in this function?